runtime/src/main/java/org/corfudb/recovery/FastSmrMapsLoader.java

```diff
@@ -183,12 +183,14 @@ public void updateStreamTails(long address, ILogData logData) {
         // On checkpoint, we also need to track the stream tail of the checkpoint
         if (isCheckPointEntry(logData)) {
             if (logData.getCheckpointType() == CheckpointEntry.CheckpointEntryType.END) {
-                streamTails.put(logData.getCheckpointedStreamId(),
-                        getStartAddressOfCheckPoint(logData));
+                streamTails.compute(logData.getCheckpointedStreamId(),
+                        (uuid, value) -> (value == null) ? getStartAddressOfCheckPoint(logData)
```

Comment:
@dahliamalkhi I don't have a concern. Initially, when I asked Zeeshan to max/ceil the stream addresses instead of the overwriting them, I also asked him to have an explicit check that will fail if addresses will go backwards, but after your example I realized that check is not necessary because addresses can go backwards, but we shouldn't throw an error instead we should just take the max.